### PR TITLE
Fix problems in Makefile

### PR DESCRIPTION
**Pull request by yangdongsheng**

### CODE REVIEW

**Comment by yangdongsheng** [comment]
Fixes https://github.com/shipyard/shipyard/issues/596

**Comment by yangdongsheng** [comment]
@asdfsx @MironWong

**Comment by tombee** [comment]
LGTM

File: Makefile [hunk 1]
```diff
@@ -10,7 +10,7 @@ clean:
 	@rm -rf controller/controller
 
 build:
-	@cd controller && godep go build -a -tags "netgo static_build" -installsuffix netgo -ldflags "-w -X github.com/shipyard/shipyard/version.GitCommit=$(COMMIT)" .
+	@cd controller && godep go build -a -tags "netgo static_build" -installsuffix netgo -ldflags "-w -X github.com/shipyard/shipyard/version.GitCommit $(COMMIT)" .
```
Comment (ehazlett) [inline Makefile]: We need the `=` as this may not work in the future (this is a Go warning upon build).

Comment (ehazlett) [inline Makefile]: To be clearer, it should not be changed from the previous.

Comment (yangdongsheng) [inline Makefile]: But, we got an error in "make image" with `=`:

# go version

go version go1.4.2 linux/amd64

Do you mean I need to upgrade my golang?

Comment (ehazlett) [inline Makefile]: You shouldn't get an error and yes you should be using Go 1.5. Thanks!

Comment (yangdongsheng) [inline Makefile]: So, can I say our Makefile is not compatible to Go version before 1.5? What about making it compatible?

**Comment by yangdongsheng** [comment]
@ehazlett  and @tombee I updated it to make it compatible to different versions of golang, please take a look

**Comment by tombee** [comment]
@yangdongsheng is there a particular reason you want it to be compatible with 1.4?

IMO we should just leave it, and ensure we build with 1.5. `Dockerfile.build` uses `golang:1.5`.

**Comment by yangdongsheng** [comment]
@tombee I am using RHEL7, and I met this error in my building :(

**Comment by tombee** [comment]
Perhaps we can look at creating a script that automates the creations of the build image from `Dockerfile.build`, and then runs `make` inside a container.

**Pull request closed without merging**